setup.py

```diff
@@ -4,34 +4,15 @@
 # Distributed under the 3-clause BSD license, see accompanying file LICENSE
 # or https://github.com/nsmith-/correctionlib for details.
 
-import sys
-
-from setuptools import setup  # isort:skip
-
-# Available at setup time due to pyproject.toml
-from pybind11.setup_helpers import Pybind11Extension  # isort:skip
-
-# Note:
-#   Sort input source files if you glob sources to ensure bit-for-bit
-#   reproducible builds (https://github.com/pybind/python_example/pull/53)
-
-if sys.platform.startswith("win"):
-    extra_compile_args = ["/Zc:__cplusplus", "/O2"]
-else:
-    extra_compile_args = ["-O3"]
-
-ext_modules = [
-    Pybind11Extension(
-        "correctionlib._core",
-        ["src/python.cc", "src/correction.cc", "src/formula_ast.cc"],
-        cxx_std=17,
-        include_pybind11=False,
-        include_dirs=["rapidjson/include", "pybind11/include", "cpp-peglib", "include"],
-        extra_compile_args=extra_compile_args,
-    ),
-]
-
+from setuptools import find_packages
+from setuptools_scm import get_version
+from skbuild import setup
 
 setup(
-    ext_modules=ext_modules,
+    packages=find_packages(where="src"),
+    package_dir={"": "src"},
+    cmake_install_dir="src",
+    cmake_args=[
+        "-DCORRECTIONLIB_VERSION={}".format(".".join(get_version().split(".")[:2]))
```

Comment:
iirc there's a way for CMake to forward this to the preprocessor, which would be nice to populate a version string and/or https://github.com/cms-nanoAOD/correctionlib/blob/master/include/correction.h#L22 inside the header.

Reply:
Good point, added, but using it for the evaluator version breaks the tests... I'll debug further later